### ISSUE TITLE
"ERROR: Error response from daemon: Bad response from Docker engine" (Docker appears to have stopped running)

### ISSUE THREAD

**Issue by phoenix-oes** [docker/desktop-linux#131]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 03f84bfe-fba8-48ba-9743-317355122474/20230315073455

### Expected behavior
Docker continues to run

### Actual behavior

Initially, everything works fine. Docker runs, Desktop works, I can open devcontainers in VS Code, it's wonderful. But.

Eventually, Docker will just stop on its own, and all Docker commands will return "ERROR: Error response from daemon: Bad response from Docker engine"

### Information

  - Linux distro: Ubuntu
  - Distro version: 20.04
  - Docker Desktop Version: v4.17.0

Fresh installs of both Ubuntu and Docker Desktop. This behavior has occurred since the beginning.

### Steps to reproduce the behavior

  1. Install Docker desktop
  2. Wait

**Comment by tomunger** [comment]
Same on 

 * Ubuntu 22.04.01 "Pro" with live patch
 * docker desktop v4.17.0

I have no containers created or running.  After several days, the docker command returns "error response from daemon" and the desktop application becomes unresponsive.  I must reboot to fix the problem.

**Comment by pmcclure07** [comment]
I have the same issue
Ubuntu 20.04.5
Docker Desktop 4.17.0
The desktop and daemon processes are still running, but all docker commands get the error.
I can kill the desktop processes and restart the desktop. Then the container shows as _exited_, and I am able to restart it...

**Comment by cheack** [comment]
Same problem here. Linux Mint 21.1 Vera. Killing all Docker-related processes helps.

**Comment by kyanitedu** [comment]
Same here, I had the Desktop running for 2 days and it is giving me the same error. I was running postgres:11.18-alpine
Ubuntu 22.04.2 LTS
Docker Desktop v4.17.0

**Comment by Liar4u** [comment]
Same:
OS: Arch Linux x86_64
Kernel: 6.2.6-zen1-1-zen
Uptime: 3 days, 4 hours, 54 mins
Docker version 23.0.1, build a5ee5b1dfc
Docker Desktop 4.17.0 (99724)

**Comment by marioheber** [comment]
Same here. I had the Desktop running:
Ubuntu 22.04.2 LTS
Docker version 23.0.1, build a5ee5b1
Docker Desktop v4.17.0

**Comment by marcello-dev** [comment]
Same here. Interestingly, the command "systemctl --user status docker-desktop" shows the service as active, but the Docker Desktop icon disappears and whenever I try to launch it, the UI appears and immediately crashes.  

The only workaround for me seems "systemctl --user restart docker-desktop" or to restart the whole machine.

Ubuntu 22.04.1
Docker version 23.0.1, build a5ee5b1
Docker Desktop v4.17.0

**Comment by aiordache** [comment]
I had a look at the diagnostics bundle provided by @phoenix-oes but it mostly contains queries/pings issued from the host components after the engine/VM has stopped responding. There are no logs from the VM after the crash and without them it is difficult to understand what it happening. 
Anybody willing to upload a bundle created shortly after the engine stopped responding to see if any error is being logged? I will try to reproduce this.

Would also help to check if the `qemu-system-x86_64` process is still running after the engine stops responding.

**Comment by pmcclure07** [comment]
My docker-desktop exited today, and `qemu-system-x86_64` is still running;

Error response from daemon: Bad response from Docker engine
$ ps -ef | grep qemu-system-x86_64
xxx   99474   99400  5 Mar21 ?        01:24:50 qemu-system-x86_64 -accel kvm -cpu host -machine q35 -m 7954 -smp 6 -kernel /opt/docker-desktop/linuxkit/kernel -append page_poison=1 vsyscall=emulate...

**Comment by aiordache** [comment]
We have fixed a networking bug since last release that may be related. Below you can find a dev build, if anyone wants to try it and confirm the issue is fixed 🙏  Thank you!

https://desktop-stage.docker.com/linux/main/amd64/102631/docker-desktop-4.18.0-amd64.deb
https://desktop-stage.docker.com/linux/main/amd64/102631/docker-desktop-4.18.0-x86_64.pkg.tar.zst

**Comment by MrSaiclops** [comment]
Updated to the latest dev build, still having the issue where I check on Docker Desktop it has reverted to a blank containers screen, as if nothing has ever been run on it before. Killing the processes and restarting Docker Desktop returns my preexisting containers to normal.

**Comment by MrSaiclops** [comment]
And mine:
```
$ docker version
Error response from daemon: Bad response from Docker engine
Client: Docker Engine - Community
 Cloud integration: v1.0.31
 Version:           23.0.1
 API version:       1.42
 Go version:        go1.19.5
 Git commit:        a5ee5b1
 Built:             Thu Feb  9 19:47:01 2023
 OS/Arch:           linux/amd64
 Context:           desktop-linux
```
```
$ docker info 
Client:  
 Context:    desktop-linux  
 Debug Mode: false  
 Plugins:   
  buildx: Docker Buildx (Docker Inc.)     
   Version:  v0.10.4     
   Path:     /usr/lib/docker/cli-plugins/docker-buildx   
  compose: Docker Compose (Docker Inc.)     
   Version:  v2.17.0-rc.1     
   Path:     /usr/lib/docker/cli-plugins/docker-compose   
  dev: Docker Dev Environments (Docker Inc.)     
   Version:  v0.1.0     
   Path:     /usr/lib/docker/cli-plugins/docker-dev   
  extension: Manages Docker extensions (Docker Inc.)     
   Version:  v0.2.18     
   Path:     /usr/lib/docker/cli-plugins/docker-extension   
  rollout: Rollout new Compose service version (Karol Musur)     
   Version:  v0.1     
   Path:     /home/theacademy/.docker/cli-plugins/docker-rollout   
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)     
   Version:  0.6.0     
   Path:     /usr/lib/docker/cli-plugins/docker-sbom   
  scan: Docker Scan (Docker Inc.)     
   Version:  v0.25.0     
   Path:     /usr/lib/docker/cli-plugins/docker-scan   
  scout: Command line tool for Docker Scout (Docker Inc.)     
   Version:  v0.6.0    
   Path:     /usr/lib/docker/cli-plugins/docker-scout  
Server: 
ERROR: Error response from daemon: Bad response from Docker engine 
errors pretty printing info
```
`qemu-system-x86_64` is still running Docker Desktop showing a blank containers screen.

**Comment by dingerkingh** [comment]
I am having the EXACT same issue. Running v4.17 as well and Ubuntu Desktop 22.04.2 LTS. Every day or two. My guess is something with 4.17 because it did not do this before I upgraded to 4.17

**Comment by TimothyLoyer** [comment]
I was having this issue as well. I [downgraded to v4.16.2](https://docs.docker.com/desktop/release-notes/#4162) yesterday and it has resolved the problem for me.

**Comment by tomunger** [comment]
I too have reverted to version 4.16.2. 

To revert, download files for the earlier version from the link provided above.  Follow docker desktop install instructions, including removing "older" versions.  I did those steps and now have 4.16.2 running.

**Comment by aiordache** [comment]
Sorry for the inconvenience, debugging this takes time. We have tracked down the problem to the VM IP lease not being renewed after 24h :grimacing:. A fix for this should be shipped in the 4.18 release coming sometimes next week.

Here are some new dev builds if anybody willing to test and confirm the fix is working:
Intel Deb https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-amd64.deb
Intel RPM https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-x86_64.rpm
Intel Arch https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-x86_64.pkg.tar.zst
Arm Deb https://desktop-stage.docker.com/linux/main/arm64/103433/docker-desktop-4.18.0-arm64.deb

**Comment by Omico** [comment]
> Sorry for the inconvenience, debugging this takes time. We have tracked down the problem to the VM IP lease not being renewed after 24h 😬. A fix for this should be shipped in the 4.18 release coming sometimes next week.
> 
> Here are some new dev builds if anybody willing to test and confirm the fix is working: Intel Deb https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-amd64.deb Intel RPM https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-x86_64.rpm Intel Arch https://desktop-stage.docker.com/linux/main/amd64/103433/docker-desktop-4.18.0-x86_64.pkg.tar.zst Arm Deb https://desktop-stage.docker.com/linux/main/arm64/103433/docker-desktop-4.18.0-arm64.deb

A huge thanks! I have installed this build and waited at least 24 hours. It is working so far. My env is Ubuntu 22.04.2 LTS x64.

**Comment by tomunger** [comment]
The new version has been running successfully for 2 days on my system.

**Comment by aiordache** [comment]
Thank you all for confirming!! 🙇 
The fix has been included in the 4.18 release that is now available.
